### ISSUE TITLE
springdoc custom paths not working as expected

### ISSUE THREAD

**Issue by knyttl** [springdoc/springdoc-openapi#957]
**Describe the bug**
We are running a service behind a proxy so that

`service-post:8080/swagger-ui.html` is routed to public address `https://host.com/services/post/swagger-ui.html`

when I open the `service-post:8080/swagger-ui.html` directly, I can correctly see the swagger docs.

To set the paths for the public address on `host.com`, I am using:

```
springdoc.api-docs.path:        /services/post/api-docs
springdoc.swagger-ui.path:      /services/post/swagger-ui.html
springdoc.swagger-ui.configUrl: /services/post/v3/api-docs/swagger-config
```

However it seems that this brakes it completely:

`/swagger-ui.html`, `/api-docs` and `/v3/api-docs/swagger-config` return 404 both for `service-post:8080/*` and `https://host.com/services/post/*`

Only thing that seems to work is `https://host.com/services/post/swagger-ui/index.html` which shows the `petstore` documentation.

Is there something I am missing?

**Comment by bnasslahsen** [comment]
@knyttl, 

 There are no known issues with the proxy. 
 All you need to do, is correctly configure your application and the proxy to serve the swagger-ui.
 It's all explained in the documentation:

 - https://springdoc.org/#how-can-i-deploy-springdoc-openapi-ui-behind-a-reverse-proxy

**Comment by knyttl** [comment]
Unfortunately the link does not solve my issue. I am not using Spring Boot (pure Spring), so I am not sure these headers would have any effect.

Can you see why the configuration above would not work?

```
springdoc.api-docs.path:        /services/post/api-docs
springdoc.swagger-ui.path:      /services/post/swagger-ui.html
springdoc.swagger-ui.configUrl: /services/post/v3/api-docs/swagger-config
```

Normally our services map correctly with the prefix and I don't see a reason why should this be a problem here. From the documentation this should normlaly work.

**Comment by bnasslahsen** [comment]
@knyttl, 

 The headers are **mandatory** and the spring-boot configuration properties `server.forward-headers-strategy=framework` or any equivalent as well.

 The properties you are not enough, for exposing the swagger-ui behind a reverse proxy.

 Not only for springodc, but for any kind of spring application behind reverse proxy.

  Make sure you add the  **reverse proxy configuration header** and all you **spring boot application configuration properties** that you have applied.

**Comment by knyttl** [comment]
Spring Boot: as I stated above, I am **not using Spring Boot**, just vanilla Spring MVC.

> The headers are mandatory and the spring-boot configuration properties server.forward-headers-strategy=framework or any equivalent as well.

Is the `server.forward-headers-strategy=framework` require by Spring Boot or the springdoc?

> Not only for springdoc-openapi, but for any kind of spring-boot application behind reverse proxy.

All services have been working for us without such headers for years. We are stripping the prefix on the `nginx` proxy before, so that:

1. nginx receives `/services/post/api-docs`
2. nginx strips the prefix and passes `/api-docs` to the service
3. services receives the request on `/api-docs`, however, when it generates the link, it must contain the prefix!

Basically I don't want to change location of the endpoint, just where it redirects or looks for config.

**Comment by bnasslahsen** [comment]
@knyttl, 

 You haven't mentioned the spring version you are using ...
 GITHUB Issues is not here for this kins of questions. Please use [the appropriate channels ](https://github.com/springdoc/springdoc-openapi/blob/master/CONTRIBUTING.adoc)for that.
